languages/js/src/Polar.ts

```diff
@@ -81,21 +81,23 @@ export class Polar {
     while (true) {
       const query = this.#ffiPolar.nextInlineQuery();
       if (query === undefined) break;
-      const { done } = new Query(query, this.#host).results.next();
+      const { results } = new Query(query, this.#host);
+      const { done } = results.next();
+      results.return();
```

Comment:
What does `results.return()` do?

Reply:
"Completes" the generator so that (ideally) the JS VM can GC the allocated memory since we have no use for anything beyond the first result

Reply:
No, it doesn't exhaust the generator by fully evaluating it -- the internal state is updated to "completed".